aiohttp/web_urldispatcher.py

```diff
@@ -414,6 +414,65 @@ def __repr__(self) -> str:
         return f"<PlainResource {name} {self._path}>"
 
 
+class GroupPlainResource(Resource):
+    def __init__(self) -> None:
+        super().__init__()
+        self._resources: Dict[str, Resource] = {}
+
+    @property
+    def canonical(self) -> str:
+        return "/"
+
+    def freeze(self) -> None:
+        self._routes.clear()
+        for resource in self._resources.values():
+            self._routes.extend(cast(Iterator[ResourceRoute], resource))
+
+    def add_prefix(self, prefix: str) -> None:
+        assert prefix.startswith("/")
+        assert not prefix.endswith("/")
+        assert len(prefix) > 1
+        old = self._resources.copy()
+        self._resources.clear()
+        for path, resource in old.items():
+            resource.add_prefix(prefix)
+            self._resources[prefix + path] = resource
+
+    async def resolve(self, request: Request) -> _Resolve:
+        allowed_methods: Set[str] = set()
+
+        resource = self._resources.get(request.rel_url.raw_path)
+        if resource is None:
+            return None, allowed_methods
+
+        return await resource.resolve(request)
+
+    def _match(self, path: str) -> Optional[Dict[str, str]]:
+        if path in self._resources:
+            return {}
+        else:
+            return None
```

Comment:
```suggestion
        return {} if path in self._resources else None
```

Could make the same change to `PlainResource._match()` while we're here.

Reply:
then the readability of the code will decrease

Reply:
Personally, I find the latter more readable, it's faster for me to see that the function returns correctly in all situations. Plus these simple 1-word literals are exactly the case these inline `if` expressions were made for.